src/analyze/analyze.c

```diff
@@ -1673,6 +1673,20 @@ static int dump_syscall_filters(int argc, char *argv[], void *userdata) {
 }
 #endif
 
+static void parsing_hint(const char *p, bool calendar, bool timestamp, bool timespan) {
+        usec_t usec;
+
+        if (calendar && calendar_spec_from_string(p, NULL) >= 0)
+                log_notice("Hint: this expression is a valid calendar specification. "
+                           "Use 'systemd-analyze calendar \"%s\"' instead?", p);
+        if (timestamp && parse_timestamp(p, &usec) >= 0)
```

Comment:
hmm, maybe patch parse_timestamp() so that it takes a NULL argument too? we don't actually care about the return value here, right?